content/posts/cloudbees-cross-team-and-dev-sec-ops.md

```diff
@@ -0,0 +1,261 @@
+  - The `sh` step will run on the agent, not the Jenkins Master, allowing you to scale across as many agents as needed for your container scans with very little impact on the performance of the Jenkins Master.
+  - Using lightweight shell scripts provide easier testing and more portability of your CD pipelines to other platforms.
+
+### Anchore Inline Scan
+Earlier this year, [Anchore](https://anchore.com/) provided some new tools and scripts to make it easier to execute Anchore scans without constantly running an Anchore Engine. The [Anchore **inline scan**](https://anchore.com/inline-scanning-with-anchore-engine/) provides the same analysis/vulnerability/policy evaluation and reporting as a statically managed Anchore engine and is used in this example to highlight how easy and fast you can add container security scanning to your own CD pipelines. However, a better long-term approach would be to stand-up your own centralized, managed and stable Anchore engine to use across all of you dev teams. The advantages of a static, always running Anchore Engine include:
+
+- Faster scans: since you don't have to wait for the Anchore engine to start-up for each job.
+- Reduced infrastructure costs: if you only do a few scans a day then this is less of an advantage as you will have a constant infrastructure cost for the static Anchore engine. But if you are doing 100s of scan per day then you will defintely realize savings with this approach.
+- More secure: as we will see in the **inline scan** example below, the Anchore `inline_scan` script requires access to a Docker daemon. And in this example we are using the [Jenkins Kubernetes plugin](https://github.com/jenkinsci/kubernetes-plugin) to provide dynamic and ephemeral agent pods for the Anchore inline scan job. A quick and dirty approach - that has a number of security implications - for providing a K8s pod agent access to the Docker daemon is to mount the Docker socket as a `volume` on the pod.
+
+*Anchore inline scan Pod* - `dockerClientPod.yml`
+```yaml
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: docker-client
+    image: gcr.io/technologists/docker-client:0.0.3
+    command: ['cat']
+    tty: true
+    volumeMounts:
+    - name: dockersock
+      mountPath: /var/run/docker.sock
+  volumes:
+  - name: dockersock
+    hostPath:
+      path: /var/run/docker.sock
+```
+
+Even though there is an [Anchore plugin for Jenkins](https://plugins.jenkins.io/anchore-container-scanner), there is no reason to install another plugin when you can accomplish the same thing with a very simple `sh` step. Again, as mentioned in my [last post here on the Technologists site](./jenkins-plugins-good-bad-ugly/) - using fewer Jenkins plugins is a **good** thing.
+
+```groovy
+      container('docker-client'){
+        sh "curl -s https://ci-tools.anchore.io/inline_scan-v0.3.3 | bash -s -- -f -b ./.anchore_policy.json -p ${containerImage}"
```

Comment:
I noticed in the browser the code block almost looks cut off at `| bash`, this suggestion lines the code block up with the copy.
```
'''groovy
container('docker-client'){
    sh "curl -s https://ci-tools.anchore.io/inline_scan-v0.3.3 | bash -s -- -f -b ./.anchore_policy.json -p ${containerImage}"
}
'''
```